### DISCUSSION TITLE
Hachyderm's 2025+ Infrastructure Plan

### DISCUSSION THREAD

**Discussion by Preskton** [hachyderm/community#589] [Infrastructure]
Hi, Hachyderm members and Fediverse friends --

We've published a [blog post](https://community.hachyderm.io/blog/2024/11/29/ensuring-hachyderms-future-improving-safety-resilience-through-strategic-placement-of-infrastructure/) about how we're thinking about Hachyderm's infrastructure given the upcoming government changes in the United States.

We've evaluated the various layers of our stack and identified a short-term and long-term set of actions we want to add to our backlog to keep Hachyderm safe, accessible, and resilient.

Because there's a lot in this post, we've opened up this Discussions section for longer-form conversations about concerns you have, items we might have missed, or questions that come up. One of our goals is to share what we're doing and learning broadly -- and this is one place that can happen, so please feel welcome to post and discuss, either in this thread or a new, focused thread on a specific subject. 💜 🐘

**Comment by bcdev-com** [comment]
Thank you, all.  I found this write up both reassuring for my personal presence on Hachyderm, but also a bunch of useful jumping off points for considering how my client's infrastructure is deployed.

The only concrete suggestion I have is that EasyDNS has been a reliable, and non-US based, Anycast DNS provider for me for a couple decades now.

**Reply by Preskton** [reply]
Thank you for the suggestion @bcdev-com! I'll add that to our list of providers we're considering; it's always good to have a recommendation, especially one that comes backed by a couple of decades of experience! 💪🏽

**Comment by enygren** [comment]
This is great!  Thanks for pulling this together.
Some thoughts that might be worth considering:

* For DNS it may make sense to split off the registrar and DNS authority functions, and to have a multi-provider setup for DNS authorities.  For example, use a registrar platform such as OpenSRS to register the domain(s) and then have a mix of authorities.  You could operate a hidden primary with the zone contents in the Core, and then use Secondary DNS services to be authoritative.  (For example Akamai/Linode Domains can be configured as a Secondary, as can others.  You can then have a mix of secondary Anycast NS records from different providers.). One downside may be that this doesn't work well with DNS-mapped names at the zone apex pointing to edge sites (eg, as "hachyderm.io" appears to be) as those are generally proprietary and don't interoperate well.

* For object storage, Akamai/Linode also has an S3-compatible object storage.   While a US company, using a mix of S3-compatible object storage at multiple providers in Frankfurt might reduce the risk from any single provider.

(Disclaimer: I work at Akamai, but comments here are intended to be personal as a user of Hachyderm and not in any official capacity.)

**Reply by Preskton** [reply]
@enygren thanks for chiming in 🎉 - your bits about DNS line up with what we've been thinking internally. At this point it's not even on paper, so plenty of work to do. But I also think splitting registrar from our NS is going to be key for maximum flexibility/redundancy.

**Comment by ljhanson** [comment]
How can I volunteer to help?

**Reply by Preskton** [reply]
Hiya, @ljhanson - details are over here ➡️ https://github.com/hachyderm/community/discussions/590. Looking forward to the convo!
